examples/llm/common/content_extractor_module.py

```diff
@@ -0,0 +1,294 @@
+                meta: typing.Optional[dict[str, typing.Any]],
+                remove_numeric_tables: typing.Optional[bool] = None,
+                valid_languages: typing.Optional[list[str]] = None,
+                encoding: typing.Optional[str] = "UTF-8",
+                id_hash_keys: typing.Optional[list[str]] = None) -> list[Document]:
+        """
+        Load a CSV file and convert it to Documents.
+
+        Parameters
+        ----------
+        file_path:
+            Path to the CSV file you want to convert.
+        meta:
+            Optional dictionary of metadata key-value pairs that you want to append to the returned document.
+        encoding:
+            Optional file encoding format, default: `UTF-8`.
+        id_hash_keys:
+            Generates the document ID from a custom list of strings that refer to the document's attributes.
+        remove_numeric_tables: bool
+            Removes numeric tables from the csv.
+        valid_languages:
+            Valid languages
+
+        Returns
+        -------
+        list[haystack.Document]
+            List of documents, 1 document per line in the CSV.
+        """
+        if not isinstance(file_path, list):
+            file_path = [file_path]
+
+        docs: list[Document] = []
+        text_column_names = ["content"]
+
+        if meta is not None:
+            text_column_names = set(meta.get("csv", {}).get("text_column_names", text_column_names))
+
+        for path in file_path:
+            df = pd.read_csv(path, encoding=encoding)
+            if len(df.columns) == 0 or (not text_column_names.issubset(set(df.columns))):
+                raise ValueError("The CSV file must either include a 'content' column or have a "
+                                 "columns specified in the meta configuration with key 'text_column_names'.")
+
+            df.fillna(value="", inplace=True)
+            df["content"] = df[text_column_names].apply(lambda x: ' '.join(map(str, x)), axis=1)
+
+            docs_dicts = df.to_dict(orient="records")
+
+            for dictionary in docs_dicts:
+                if meta:
+                    dictionary["meta"] = meta
+                if id_hash_keys:
+                    dictionary["id_hash_keys"] = id_hash_keys
+                docs.append(Document.from_dict(dictionary))
+
+        return docs
+
+
+def get_file_meta(open_file: fsspec.core.OpenFile) -> FileMeta:
+    """
+    Extract file metadata from the given open file.
+
+    Parameters
+    ----------
+    open_file: fsspec.core.OpenFile
+        OpenFile object
+
+    Returns
+    -------
+    FileMeta
+        Returns FileMeta instance.
+    """
+    try:
+        file_path = open_file.path
+        file_name = file_path.split('/')[-1]
+        split_result = file_name.lower().rsplit('.', 1)
+
+        if len(split_result) > 1:
+            _, file_type = split_result
+        else:
+            file_type = "none"
+
```

Comment:
I would probable do:
```suggestion
        file_name = os.path.basename(file_path)
        _, file_type = os.path.splitext(file_name)

if len(file_type) > 0:
    file_type = file_type.lstrip('.')
else:
    file_type = 'none'
```